src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +104,64 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
         commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def runs = project.nunit.getTestInputAsList(getWhereFilters())
+
+        GParsPool.withPool {
+            runs.eachParallel {
+                def fileName = getRandomFileName()
+                logger.info("Filename generated for the \'$it\' input was \'$fileName\'")
+
+                def execArgs = project.nunit.buildCommandArgs(it, new File(intermediateReportsPath, fileName + ".xml"))
```

Comment:
this file seems to be the coverage report file (as specified below, and used in the merge process), seems strange to pass it to nunit as the nunit report path (should be different files), no ?

Reply:
Good point! 👍